src/instance_launcher.ts

```diff
@@ -80,18 +80,46 @@ export default class InstanceLauncher {
                 const actualScaleUpQuantity =
                     Math.min(group.scalingOptions.maxDesired, group.scalingOptions.desiredCount) - count;
                 const scaleDownProtected = await this.instanceGroupManager.isScaleDownProtected(group.name);
-                await this.cloudManager.scaleUp(ctx, group, count, actualScaleUpQuantity, scaleDownProtected);
+                const scaleUpCount = await this.cloudManager.scaleUp(
+                    ctx,
+                    group,
+                    count,
+                    actualScaleUpQuantity,
+                    scaleDownProtected,
+                );
 
-                await this.audit.saveLauncherActionItem(groupName, {
-                    timestamp: Date.now(),
-                    actionType: 'scaleUp',
-                    count: count,
-                    desiredCount: group.scalingOptions.desiredCount,
-                    scaleQuantity: actualScaleUpQuantity,
-                });
+                if (scaleUpCount > 0) {
+                    await this.audit.saveLauncherActionItem(groupName, {
+                        timestamp: Date.now(),
+                        actionType: 'scaleUp',
+                        count: count,
+                        desiredCount: group.scalingOptions.desiredCount,
+                        scaleQuantity: scaleUpCount,
+                    });
+
+                    // increment launched instance stats for the group
+                    instancesLaunchedCounter.inc({ group: group.name }, scaleUpCount);
 
-                // increment launched instance stats for the group
-                instancesLaunchedCounter.inc({ group: group.name }, actualScaleUpQuantity);
+                    // check if scale up count didn't meet requested quantity, error if so
+                    if (scaleUpCount != actualScaleUpQuantity) {
+                        ctx.logger.error(
+                            `[Launcher] Scaling failed to launch requested new instances for group ${groupName} with ${count} instances.`,
+                            { scaleUpRequested: actualScaleUpQuantity, scaleUpActual: scaleUpCount },
+                        );
+                        instanceErrorsCounter.inc({ group: group.name });
+                        throw new Error(
```

Comment:
This error is caught locally and so the instanceErrorsCounter will be incremented twice; we also increment the instanceErrorsCounter on catch (err)

Reply:
Agreed, good catch.